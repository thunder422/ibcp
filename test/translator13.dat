##########################################################
#  Translator Test #13: Negative Constants (2011-03-27)  #
##########################################################

A=---B
A=-1.0
A%=--1
A=B-1.5
A=B+-1.5
A=B+--1.5
PRINT-1;--1;---1;- 1;
A=- 1.0 + -.001 + -0.12
A%=-1 + - 2
A=-1.2E-23 + -1.2e+23 + -1.2e23
A=- 1.2E-23 + - 1.2e+23 + - 1.2e23
A% = NOT -B%

# negative integer limit
A% = NOT -2147483648
A% = - NOT 123

# integer limit test
A% = -2147483649 
A = B^-C + B*-C
A = -B^C + -B^-C
A = -B*C + -B*-C

# TODO check with recreator
A = (-B^NOT C) + -B*NOT C
A = -B^NOT C + -B*NOT C
