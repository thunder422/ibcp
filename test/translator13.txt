
Input: A=---B
Output: A<ref> B -U -U -U Assign 

Input: A=-1.0
Output: A<ref> -1.0 Assign 

Input: A%=--1
Output: A%<ref> -1% -U% Assign% 

Input: A=B-1.5
Output: A<ref> B 1.5 - Assign 

Input: A=B+-1.5
Output: A<ref> B -1.5 + Assign 

Input: A=B+--1.5
Output: A<ref> B -1.5 -U + Assign 

Input: PRINT-1;--1;---1;- 1;
Output: -1% PrintInt -1% -U% PrintInt -1% -U% -U% PrintInt 1% -U% PrintInt ; 

Input: A=- 1.0 + -.001 + -0.12
Output: A<ref> 1.0 -U -.001 + -0.12 + Assign 

Input: A%=-1 + - 2
Output: A%<ref> -1% 2% -U% +% Assign% 

Input: A=-1.2E-23 + -1.2e+23 + -1.2e23
Output: A<ref> -1.2E-23 -1.2e+23 + -1.2e23 + Assign 

Input: A=- 1.2E-23 + - 1.2e+23 + - 1.2e23
Output: A<ref> 1.2E-23 -U 1.2e+23 -U + 1.2e23 -U + Assign 

Input: A% = NOT -B%
Output: A%<ref> B% -U% NOT Assign% 

Input: A% = NOT -2147483648
Output: A%<ref> -2147483648% NOT Assign% 

Input: A% = - NOT 123
Output: A%<ref> 123% NOT -U% Assign% 

Input: A% = -2147483649 
            ^^^^^^^^^^^-- expected valid integer constant

Input: A = B^-C + B*-C
Output: A<ref> B C -U ^ B C -U * + Assign 

Input: A = -B^C + -B^-C
Output: A<ref> B C ^ -U B C -U ^ -U + Assign 

Input: A = -B*C + -B*-C
Output: A<ref> B -U C * B -U C -U * + Assign 

Input: A = (-B^NOT C) + -B*NOT C
Output: A<ref> B C CvtInt NOT ^* -U')' B -U C CvtInt NOT *%2 + Assign 

Input: A = -B^NOT C + -B*NOT C
Output: A<ref> B C B -U C CvtInt NOT *%2 + CvtInt NOT ^* -U Assign 

