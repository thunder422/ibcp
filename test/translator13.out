
Input: A=---B
Output: A = ---B 

Input: A=-1.0
Output: A = -1.0 

Input: A%=--1
Output: A% = --1 

Input: A=B-1.5
Output: A = B - 1.5 

Input: A=B+-1.5
Output: A = B + -1.5 

Input: A=B+--1.5
Output: A = B + --1.5 

Input: PRINT-1;--1;---1;- 1;
Output: PRINT -1; --1; ---1; - 1; 

Input: A=- 1.0 + -.001 + -0.12
Output: A = - 1.0 + -.001 + -0.12 

Input: A%=-1 + - 2
Output: A% = -1 + - 2 

Input: A=-1.2E-23 + -1.2e+23 + -1.2e23
Output: A = -1.2E-23 + -1.2e+23 + -1.2e23 

Input: A=- 1.2E-23 + - 1.2e+23 + - 1.2e23
Output: A = - 1.2E-23 + - 1.2e+23 + - 1.2e23 

Input: A% = NOT -B%
Output: A% = NOT -B% 

Input: A% = NOT -2147483648
Output: A% = NOT -2147483648 

Input: A% = - NOT 123
Output: A% = -NOT 123 

Input: A% = -2147483649 
            ^^^^^^^^^^^-- expected valid integer constant

Input: A = B^-C + B*-C
Output: A = B ^ -C + B * -C 

Input: A = -B^C + -B^-C
Output: A = -B ^ C + -B ^ -C 

Input: A = -B*C + -B*-C
Output: A = -B * C + -B * -C 

Input: A = (-B^NOT C) + -B*NOT C
Output: A = (-B ^ NOT C) + -B * NOT C 

Input: A = -B^NOT C + -B*NOT C
Output: A = -B ^ NOT C + -B * NOT C 

